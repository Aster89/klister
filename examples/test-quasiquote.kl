#lang "prelude.kl"

[import [shift "prelude.kl" 1]]
[import [shift "defun.kl" 1]]

[define thing [quote nothing]]

[example thing]
[example [quasiquote thing]]
[example [quasiquote [unquote thing]]]

[example [quasiquote [thing]]]
[example [quasiquote [[unquote thing]]]]

[example [quasiquote [list-syntax [[unquote thing] thing] thing]]]

[example [quasiquote [list-syntax [[unquote thing] thing ()] thing]]]

[example [quasiquote (thing [unquote thing] thing)]]

(example (quasiquote/loc thing (thing ,thing)))
