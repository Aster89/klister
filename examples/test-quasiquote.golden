#[test-quasiquote.kl:6.22-6.29]<nothing> : Syntax
#[test-quasiquote.kl:9.22-9.27]<thing> : Syntax
#[test-quasiquote.kl:6.22-6.29]<nothing> : Syntax
#[test-quasiquote.kl:12.22-12.29]<(thing)> : Syntax
#[test-quasiquote.kl:13.22-13.39]<(nothing)> : Syntax
#[test-quasiquote.kl:15.22-15.65]<(list-syntax (nothing thing) thing)> : Syntax
#[test-quasiquote.kl:17.22-17.68]<(list-syntax (nothing thing ()) thing)> : Syntax
#[test-quasiquote.kl:19.22-19.51]<(thing nothing thing)> : Syntax
#[test-quasiquote.kl:6.22-6.29]<(thing nothing)> : Syntax
